android/modules/ui/src/java/ti/modules/titanium/ui/widget/tabgroup/TiUIBottomNavigation.java

```diff
@@ -0,0 +1,553 @@
+/**
+ * Titanium SDK
+ * Copyright TiDev, Inc. 04/07/2022-Present. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+package ti.modules.titanium.ui.widget.tabgroup;
+
+import android.annotation.SuppressLint;
+import android.app.Activity;
+import android.content.res.ColorStateList;
+import android.graphics.Color;
+import android.graphics.drawable.Drawable;
+import android.graphics.drawable.RippleDrawable;
+import android.os.Build;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.Window;
+import android.widget.FrameLayout;
+import android.widget.RelativeLayout;
+
+import androidx.annotation.NonNull;
+import androidx.core.graphics.ColorUtils;
+
+import com.google.android.material.badge.BadgeDrawable;
+import com.google.android.material.bottomnavigation.BottomNavigationMenuView;
+import com.google.android.material.bottomnavigation.BottomNavigationView;
+import com.google.android.material.navigation.NavigationBarView;
+import com.google.android.material.shape.CornerFamily;
+import com.google.android.material.shape.MaterialShapeDrawable;
+import com.google.android.material.shape.ShapeAppearanceModel;
+
+import org.appcelerator.titanium.TiApplication;
+import org.appcelerator.titanium.TiBaseActivity;
+import org.appcelerator.titanium.TiC;
+import org.appcelerator.titanium.TiDimension;
+import org.appcelerator.titanium.proxy.TiViewProxy;
+import org.appcelerator.titanium.util.TiConvert;
+import org.appcelerator.titanium.util.TiIconDrawable;
+import org.appcelerator.titanium.util.TiRHelper;
+import org.appcelerator.titanium.util.TiUIHelper;
+import org.appcelerator.titanium.view.TiUIView;
+
+import java.util.ArrayList;
+
+import ti.modules.titanium.ui.TabGroupProxy;
+import ti.modules.titanium.ui.TabProxy;
+
+/**
+ * TabGroup implementation using BottomNavigationView as a controller.
+ */
+public class TiUIBottomNavigation extends TiUIAbstractTabGroup implements BottomNavigationView.OnItemSelectedListener
+{
+
+	protected final static String TAG = "TiUIBottomNavigation";
+	static int id_layout = 0;
+	static int id_content = 0;
+	static int id_bottomNavigation = 0;
+	private int currentlySelectedIndex = -1;
+	private ArrayList<MenuItem> mMenuItemsArray = new ArrayList<MenuItem>();
+	private RelativeLayout layout = null;
+	private FrameLayout centerView;
+	private BottomNavigationView bottomNavigation;
+	private ArrayList<Object> tabsArray = new ArrayList<Object>();
+
+	public TiUIBottomNavigation(TabGroupProxy proxy, TiBaseActivity activity)
+	{
+		super(proxy, activity);
+	}
```

Comment:
Unnecessary super-call?

Reply:
I've used the same syntax as in the current implementation:
https://github.com/tidev/titanium-sdk/blob/562925164017ff84c94c6bd15b324a03fd2d138b/android/modules/ui/src/java/ti/modules/titanium/ui/widget/tabgroup/TiUIBottomNavigationTabGroup.java#L63
but I can check if it is needed, working on a project that uses M3 so I'll see the result right way

Reply:
Will complain right away that a super() is needed

Reply:
![Bildschirmfoto_20250312_221124](https://github.com/user-attachments/assets/73d9e1e6-37ac-4292-9eaf-847888ae48d9)

Needed because of the TiUIAbstractTabGroup setup

Reply:
That should really be more cleaned up from the beginning, but then it's fine for this one.